pkg/events/event_system_test.go

```diff
@@ -205,3 +205,34 @@ func TestRingBufferCapacity(t *testing.T) {
 	capacity = getRingBufferCapacity()
 	assert.Equal(t, uint64(configs.DefaultEventRingBufferCapacity), capacity)
 }
+
+func TestTruncateEventMessage(t *testing.T) {
+	testCases := []struct {
+		name           string
+		message        string
+		expectedLength int
+	}{
+		{
+			name:           "message length less than 1024 characters",
+			message:        "Lorem ipsum",
+			expectedLength: 11,
+		},
+		{
+			name:           "message length exactly 1024 characters",
+			message:        "Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos. Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus duis convallis. Tempus leo eu aenean sed diam urna tempor. Pulvinar vivamus fringilla lacus nec metus bibendum egestas. Iaculis massa nisl malesuada lacinia integer nunc posuere. Ut hendrerit semper vel class aptent taciti sociosqu. Ad litora torquent per conubia nostra inceptos himenaeos. Lorem ipsum dolor sit amet consectetur adipiscing elit. Quisque faucibus ex sapien vitae pellentesque sem placerat. In id cursus mi pretium tellus d",
```

Comment:
I suggest a more simplistic approach. Just use a helper function which generates a string (eg. "xxxxxx") with the specifed number of characters (`getTestString(5)` --> "xxxxx"). Also, compare the contents as well, not just the length.